pkg/db/migration/convert/convert000033/convert.go

```diff
@@ -0,0 +1,195 @@
+package convert000033
+
+import (
+	"fmt"
+	"strings"
+
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+	"gorm.io/gorm"
+
+	"github.com/instill-ai/pipeline-backend/pkg/datamodel"
+	"github.com/instill-ai/pipeline-backend/pkg/db/migration/convert"
+)
+
+const batchSize = 100
+
+type ConvertWebFields struct {
+	convert.Basic
+}
+
+var webCompNames []string
+
+func (c *ConvertWebFields) Migrate() error {
+	if err := c.migratePipeline(); err != nil {
+		return err
+	}
+
+	if err := c.migratePipelineRelease(); err != nil {
+		return err
+	}
+
+	if len(webCompNames) > 0 {
+		return fmt.Errorf("web components' output are not updated: %v", webCompNames)
+	}
+
+	return nil
+}
+
+func (c *ConvertWebFields) migratePipeline() error {
+	pipelines := make([]*datamodel.Pipeline, 0, batchSize)
+	return c.DB.Select("uid", "recipe_yaml", "recipe").FindInBatches(&pipelines, batchSize, func(tx *gorm.DB, _ int) error {
+		for _, p := range pipelines {
+			isRecipeUpdated := false
+			l := c.Logger.With(zap.String("pipelineUID", p.UID.String()))
+
+			if p.Recipe != nil {
```

Comment:
How about detecting `if p.Recipe == nil { continue }`? Same for `!isRecipeUpdated`. Try to keep the indented code for non-happy paths.